### DISCUSSION TITLE
Ally Error for Inline Scope in Redirect.

### DISCUSSION THREAD

**Discussion by eokwukwe** [adonisjs/core#4615] [Help]
This issue is related to the [one](https://discord.com/channels/423256550564691970/1245007679122964540/1245007679122964540) on discord. Here is the link to the discord post for [this](https://discord.com/channels/423256550564691970/1251135591849000971) 

However, my use case is somewhat different as I want to use it not only for sign up or log in, I also want to use it to request permission for other google services and will need to specify the scope inline in the redirect. I use it for authentication but also use it to allow authenticated users give oauth permission to services like YouTube, Drive e.t.c. and this is meant to be a separate process outside the sign up or log in.

When I define the scope inside the ally configuration, it works okay, but specifying the scope inline  in the `redirect` returns the error `OAuth 2 parameters can only have a single value: scope` 

However, what I did to handle the different use cases was to define two separate `ally` google configurations, one for normal authentication and one for another service. Here is a sample solution that I intend to use
```js
const allyConfig = defineConfig({
  google: services.google({
    clientId: env.get('GOOGLE_CLIENT_ID'),
    clientSecret: env.get('GOOGLE_CLIENT_SECRET'),
    callbackUrl: `${env.get('APP_URL')}/auth/google/callback`,
  }),
  google_youtube: services.google({
    clientId: env.get('GOOGLE_CLIENT_ID'),
    clientSecret: env.get('GOOGLE_CLIENT_SECRET'),
    callbackUrl: `${env.get('APP_URL')}/auth/youtube/callback`,

    accessType: 'offline',
    scopes: ['https://www.googleapis.com/auth/youtube.upload'],
  }),
})
```

So in my controllers I can do:
`const google = ally.use('google')` for user authentication and
`const google = ally.use('google_youtube')` for the YouTube service

The means that if I want to add permission request for another service I will have to manually add it to the config again.

Also, there seem to be no way to define `state` for the callback url.

Can the awesome 🔥 adonisjs team look into this or is there something I am missing.

**Comment by thetutlage** [comment]
How do you define it inline? Can you share that code?

**Reply by eokwukwe** [reply]
@thetutlage, this is how I defined the scopes inline according to the doc:

```js
 async redirect({ ally }: HttpContext) {
    return ally
      .use('google')
      .redirect((request) =>
        request.scopes(['userinfo.email', 'userinfo.profile'])
      )
  }
```
